dask_sql/context.py

```diff
@@ -102,6 +102,9 @@ def __init__(self, logging_level=logging.INFO):
         self.context = DaskSQLContext(self.catalog_name, self.schema_name)
         self.context.register_schema(self.schema_name, DaskSchema(self.schema_name))
 
+        if dask_config.get("sql.dynamic_partition_pruning"):
+            self.context.apply_dynamic_partition_pruning()
+
```

Comment:
Sometime configs can be passed through the `c.sql` method as well https://github.com/dask-contrib/dask-sql/blob/main/dask_sql/context.py#L468. 

Unless you need this during context initialization, it might make sense to set this variable somewhere in `_get_ral` before the query is parsed.

Reply:
Thanks! Done with https://github.com/dask-contrib/dask-sql/pull/1102/commits/182fe8ed938ac71e6fefe99e8ebe5f59188a08e9 - with this, DPP can be configured either globally (default True) or on a per-query basis with the `config_options` parameter.